### PR TITLE
Logging4 2

### PR DESCRIPTION
**Pull request by kdemski**

### CODE REVIEW

File: app/src/main/java/com/aeriosense/projecttalon/controller/DataLogger.java [hunk 1]
```diff
@@ -0,0 +1,498 @@
...
+    private Runnable runnable = new Runnable() {
+        @Override
+        public void run() {
+            appendDataLog();
+            handler.postDelayed(this, 1000);
+        }
+    };
+
+
+    private void initAirlink(){
```
Comment (owen927) [inline app/src/main/java/com/aeriosense/projecttalon/controller/DataLogger.java]: looks like there is no djiKey for these data?

File: app/src/main/java/com/aeriosense/projecttalon/missionManager/TimeLineMissionControl.java [hunk 2]
```diff
@@ -214,12 +214,18 @@ private void updateTimelineStatus(@Nullable TimelineElement element, TimelineEve
...
                     ? ""
                     : "Failed:"
                     + error.getDescription()));
         }
+        DataLogger.timeLineStringOrganizer(runningInfo);
+
+    }
 
+    public static String getRunningInfo(){
+        return runningInfo;
```
Comment (owen927) [inline app/src/main/java/com/aeriosense/projecttalon/missionManager/TimeLineMissionControl.java]: move this method to the top by all the other getters and setters

File: app/src/main/java/com/aeriosense/projecttalon/controller/MainActivity.java [hunk 3]
```diff
@@ -650,7 +650,7 @@ private void updateTelemetry() {
      * setRunningInfo gets currentAFS timeline mission state and output to textview
```
Comment (owen927) [inline app/src/main/java/com/aeriosense/projecttalon/controller/MainActivity.java]: remember to change the comment name as well

File: app/src/main/java/com/aeriosense/projecttalon/controller/DataLogger.java [hunk 4]
```diff
@@ -0,0 +1,498 @@
...
+                timeLineStringList.add(runningInfo);
+            } else {
+                if (!timeLineStringList.contains(runningInfo)) {
+                    timeLineStringList.add(runningInfo);
+                }
+            }
+        }
+    }
+
+    private void resetTimeLineLog(){
```
Comment (owen927) [inline app/src/main/java/com/aeriosense/projecttalon/controller/DataLogger.java]: a little confused about whats happening in this function, do you mind explaining whats being done here?

File: app/src/main/java/com/aeriosense/projecttalon/controller/DataLogger.java (hunk 1 above)
Comment (kdemski) [inline app/src/main/java/com/aeriosense/projecttalon/controller/DataLogger.java]: there is, UPLINK_SIGNAL_QUALITY and downlink... in djiKey, but they both yield zero as far as I can tell.

File: app/src/main/java/com/aeriosense/projecttalon/missionManager/TimeLineMissionControl.java (hunk 2 above)
Comment (kdemski) [inline app/src/main/java/com/aeriosense/projecttalon/missionManager/TimeLineMissionControl.java]: done

File: app/src/main/java/com/aeriosense/projecttalon/controller/MainActivity.java (hunk 3 above)
Comment (kdemski) [inline app/src/main/java/com/aeriosense/projecttalon/controller/MainActivity.java]: done

File: app/src/main/java/com/aeriosense/projecttalon/controller/DataLogger.java (hunk 4 above)
Comment (kdemski) [inline app/src/main/java/com/aeriosense/projecttalon/controller/DataLogger.java]: I'll put a comment in

File: app/src/main/java/com/aeriosense/projecttalon/missionManager/TimeLineMissionControl.java [hunk 5]
```diff
@@ -32,12 +32,13 @@
...
 import com.aeriosense.projecttalon.controller.TalonApplication;
 import com.aeriosense.projecttalon.missionLoader.FlightPlanData;
+import com.aeriosense.projecttalon.controller.DataLogger;
 
 public class TimeLineMissionControl  {
 
     private MissionControl missionControl;
 
-    public String runningInfo = " ";
+    private static String runningInfo = " ";
```
Comment (owen927) [inline app/src/main/java/com/aeriosense/projecttalon/missionManager/TimeLineMissionControl.java]: why is this set to static?

File: app/src/main/java/com/aeriosense/projecttalon/controller/DataLogger.java [hunk 6]
```diff
@@ -0,0 +1,604 @@
...
+
+        }
+
+    }
+
+    /**
+     *
+     * @return the time in h:m:s format current timezone
+     */
+    private String getTime(){
```
Comment (owen927) [inline app/src/main/java/com/aeriosense/projecttalon/controller/DataLogger.java]: better name for these two get time functions would be appreciated, something getTimeHMS()?

Comment (owen927) [inline app/src/main/java/com/aeriosense/projecttalon/controller/DataLogger.java]: or getTimeWithSeperation

File: app/src/main/java/com/aeriosense/projecttalon/controller/DataLogger.java [hunk 7]
```diff
@@ -0,0 +1,604 @@
...
+                String.valueOf(c.getTime().getMinutes())  + ":" +
+                String.valueOf(c.getTime().getSeconds());
+        return time;
+    }
+
+    /**
+     * get time with '.' as the hour minute second seperator
+     * @return
+     */
+    private String getTime2(){
```
Comment (owen927) [inline app/src/main/java/com/aeriosense/projecttalon/controller/DataLogger.java]: see above

File: app/src/main/java/com/aeriosense/projecttalon/missionManager/TimeLineMissionControl.java (hunk 5 above)
Comment (kdemski) [inline app/src/main/java/com/aeriosense/projecttalon/missionManager/TimeLineMissionControl.java]: changed to non-static

File: app/src/main/java/com/aeriosense/projecttalon/controller/DataLogger.java (hunk 6 above)
Comment (kdemski) [inline app/src/main/java/com/aeriosense/projecttalon/controller/DataLogger.java]: altered

File: app/src/main/java/com/aeriosense/projecttalon/controller/DataLogger.java (hunk 7 above)
Comment (kdemski) [inline app/src/main/java/com/aeriosense/projecttalon/controller/DataLogger.java]: altered

**Pull request merged**